src/compiler/transformers/es2015.ts

```diff
@@ -3636,11 +3652,15 @@ namespace ts {
             // Visit the tag expression
             const tag = visitNode(node.tag, visitor, isExpression);
 
-            // Allocate storage for the template site object
-            const temp = createTempVariable(hoistVariableDeclaration);
+            // Allocate storage for the template site object if we're in a module.
+            // In the global scope, any variable we currently generate could conflict with
+            // variables from outside of the current compilation.
+            const temp = isExternalModule(currentSourceFile) ? createTempVariable(recordTaggedTemplateString) : undefined;

```

Comment:
Could this variable be created closer to its use, or is there some side effect I'm missing?
If you do move it next to its use, you could just replace `!temp ? ...` with `!isExternalModule(currentSourceFile) ? ...`